### PR TITLE
feature/ ci cd

### PR DESCRIPTION
**Pull request by Merciful12**

### CODE REVIEW

File: scripts/tests.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+function testPIURI(changedFilePaths, logger) {
+  let countErrors = 0
+  changedFilePaths.forEach((filePath) => {
+    logger.info(`filePath: ${filePath}`)
+
+    const parsedFilePath = path.parse(filePath)
+    const dirs = parsedFilePath.dir.split('/').slice(3) // strip site/content/protocols
+    const [name, version] = dirs
+
+    if (!testStructure(dirs)) {
```
Comment (ashcherbakov) [inline scripts/tests.js:52]: Should we do this check before we do `const [name, version] = dirs` assignment?

File: scripts/tests.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      logger.error('should have correct path')
```
Comment (ashcherbakov) [inline scripts/tests.js]: Is it the message to be shown in UI for the user? If so, let's make it more detailed. In particular, mention what is a `correct` path (for example: "a new protocol must be in <protocol_name>/<protocol_version> folder").

File: scripts/tests.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      countErrors++
+    }
+
+    if (!testName(name)) {
+      logger.error('should have correct name folder')
+      countErrors++
+    }
+
+    if (!testVersion(version)) {
+      logger.error('should have correct verstion folder')
```
Comment (ashcherbakov) [inline scripts/tests.js]: 1) Typo: verstion -> version
2) Please specify what we understand by a "correct" version

File: scripts/tests.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      countErrors++
+    }
+
+    if (!testFileName(parsedFilePath.base)) {
+      logger.error(
+        'should have correct markdowm file name ' + parsedFilePath.base,
```
Comment (ashcherbakov) [inline scripts/tests.js]: 1) typo: markdowm -> markdown
2) Please specify what we understand by a "correct" markdown

File: scripts/tests.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    if (!testName(name)) {
+      logger.error('should have correct name folder')
+      countErrors++
+    }
+
+    if (!testVersion(version)) {
+      logger.error('should have correct verstion folder')
+      countErrors++
+    }
+
```
Comment (ashcherbakov) [inline scripts/tests.js:69]: Do we check that name and version should be unique?

File: .remarkrc [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      "remark-frontmatter",
+      [
+        "yaml"
+      ]
+    ],
+    [
+      "remark-frontmatter-validator",
+      [
+        "error",
+        {
```
Comment (ashcherbakov) [inline .remarkrc:15]: 1) Do we check other metadata such as Authors and PIURI?
2) Do we check a correctness of the PIURI value?

File: .remarkrc [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    ["remark-lint-no-dead-urls", ["error"]],
+    ["remark-lint-no-html", ["error"]],
+    [
+      "remark-frontmatter",
+      [
+        "yaml"
+      ]
+    ],
+    [
+      "remark-frontmatter-validator",
```
Comment (ashcherbakov) [inline .remarkrc:12]: Do we check a correctness of a header, that it has <Protocol Name> <Protocol version> format? See https://hackmd.io/d6BQ4yxXQZa9y9ppEruP4Q?view

File: .remarkrc [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      [
```
Comment (ashcherbakov) [inline .remarkrc:13]: Also it looks like we need to check for a Summary as it's shown in the search results

File: scripts/tests.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    const parsedFilePath = path.parse(filePath)
+    const dirs = parsedFilePath.dir.split('/').slice(3) // strip site/content/protocols
+    const [name, version] = dirs
+
+    if (!testStructure(dirs)) {
+      logger.error('should have correct path')
+      countErrors++
+    }
+
+    if (!testName(name)) {
```
Comment (ashcherbakov) [inline scripts/tests.js:60]: Do we check that the folder matches the protocol name as defined in the Header and in metadata?

File: scripts/tests.js [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+      logger.error('should have correct name folder')
+      countErrors++
+    }
+
+    if (!testVersion(version)) {
```
Comment (ashcherbakov) [inline scripts/tests.js:65]: Do we check that the folder matches the protocol version as defined in the Header and in metadata?

File: scripts/tests.js (hunk 9 above)
Comment (ashcherbakov) [inline scripts/tests.js:60]: Also we need to check that `Protocol names never differ only by case or punctuation...` (see https://hackmd.io/s8LmceVRRf-23iTKCy7X8A#Best-practices-and-CICD)

File: scripts/tests.js (hunk 5 above)
Comment (Merciful12) [reply scripts/tests.js:69]: I think it's impossible to make not unique folders structure. What do you think?

File: scripts/tests.js (hunk 10 above)
Comment (Merciful12) [reply scripts/tests.js:65]: Version should not be placed in metadata manually. It will be added on build time (extracted from second folder name)

**Pull request merged**